src/internal/connector/data_collections.go

```diff
@@ -138,13 +121,23 @@ func (gc *GraphConnector) ProduceBackupCollections(
 			return nil, nil, err
 		}
 
-		gc.incrementMessagesBy(len(colls))
```

Comment:
This may fail for the current branch.  not planning to merge to main until the next branch folds in.